Sources/CodeGen/CXX/CXXTranspiler.swift

```diff
@@ -3,56 +3,62 @@ import Core
 import FrontEnd
 import Utils
 
-/// A Val to C++ transpiler.
+/// The conversion of typed Val module AST into corresponding C++ AST.
 public struct CXXTranspiler {
 
-  /// The program being transpiled.
-  let program: TypedProgram
+  /// The Val typed nodes.
+  ///
+  /// Used in a few cases in which we need to obtain the typed nodes from the corresponding Node IDs,
+  /// and to make general queries in the Val AST.
```

Comment:
```suggestion
  /// This property is used when we need access to the contents of a node from its ID.
```
But perhaps you don' need this additional precision at all. Programs are consistently used to access the contents of the AST everywhere in the compiler. It's obvious that, if the property exists, it is there for "the cases in which we need to obtain the typed nodes".